src/redteam/commands/discover.ts

```diff
@@ -0,0 +1,318 @@
+    .addOption(
+      new Option(
+        '--turns <turns>',
+        'A maximum number of turns to run the discovery process. Lower is faster but less accurate.',
+      ).argParser(Number.parseInt),
+    )
+
+    .option('--env-path <path>', 'Path to a custom .env file')
+    .option('-v, --verbose', 'Show debug logs')
+    .action(async (rawArgs: Args) => {
+      // Validate the arguments:
+      const { success, data: args, error } = ArgsSchema.safeParse(rawArgs);
+      if (!success) {
+        logger.error('Invalid options:');
+        error.issues.forEach((issue) => {
+          logger.error(`  ${issue.path.join('.')}: ${issue.message}`);
+        });
+        process.exitCode = 1;
+        return;
+      }
+
+      // Set up the environment:
+      setupEnv(args.envPath);
+
+      // Set up logging:
+      if (args.verbose) {
+        setLogLevel('debug');
+      }
+
+      // Record telemetry:
+      telemetry.record('command_used', {
+        name: 'redteam discover',
+      });
+
+      // Always use remote for discovery
+      cliState.remote = true;
+
+      let config: UnifiedConfig | null = null;
+      // Although the providers/targets property supports multiple values, Redteaming only supports
+      // a single target at a time.
+      let target: ApiProvider | undefined = undefined;
+      // Fallback to the default config path:
+      const fallbackConfigPath = path.join(process.cwd(), 'promptfooconfig.yaml');
+
+      // If user provides a config, read the target from it:
+      if (args.config) {
+        // Validate that the config is a valid path:
+        if (!fs.existsSync(args.config)) {
+          throw new Error(`Config not found at ${args.config}`);
+        }
+
+        config = await readConfig(args.config);
+
+        if (!config) {
+          throw new Error(`Config is invalid at ${args.config}`);
+        }
+
+        if (!config.providers) {
+          throw new Error('Config must contain at least one target or provider');
+        }
+
+        const providers = await loadApiProviders(config.providers);
+        target = providers[0];
+      }
+      // If the target flag is provided, load it from Cloud:
+      else if (args.target) {
+        // Let the internal error handling bubble up:
+        const providerOptions = await getProviderFromCloud(args.target);
+        target = await loadApiProvider(providerOptions.id, { options: providerOptions });
+      }
+      // Check the current working directory for a promptfooconfig.yaml file:
+      else if (fs.existsSync(fallbackConfigPath)) {
+        config = await readConfig(fallbackConfigPath);
+
+        if (!config) {
+          throw new Error(`Config is invalid at ${fallbackConfigPath}`);
+        }
+
+        if (!config.providers) {
+          throw new Error('Config must contain at least one target or provider');
+        }
+
+        const providers = await loadApiProviders(config.providers);
+        target = providers[0];
```

Comment:
Versions of this exist in other parts of the code base. Would be better if you could re-use